src/mongocxx/database.cpp

```diff
@@ -157,5 +164,16 @@ collection database::collection(stdx::string_view name) const {
     return mongocxx::collection(*this, name);
 }
 
+const database::impl& database::get_impl() const {
+    if(!_impl) {
+        throw operation_exception{make_error_code(error_code::k_database_uninitialized)};
```

Comment:
I wasn't sure about this, since I didn't know whether all methods that call get_impl would want to throw the same kind of exception. But it makes sense that it is a logical error in all cases, so I like the idea of a mongocxx::logic_error.